scripts/cleanup_old_vms.py

```diff
@@ -44,167 +35,181 @@ def parse_cmd_line():
-                    vm_config_datastore = provider.get_vm_config_files_path(vm_name)
-                    datastore_url = provider.get_vm_datastore_path(vm_name, vm_config_datastore)
-                    vm_creation_time = get_vm_config_modified_time(hostname, vm_name,
-                                                                   datastore_url, provider_key)
-                else:
-                    vm_creation_time = provider.vm_creation_time(vm_name)
-
-                if vm_creation_time is False:
-                    # This VM must have some problem, include in report even though we can't delete
-                    status = provider.vm_status(vm_name)
-                    deleted_vms_list.append([provider_key,
-                                             vm_name,
-                                             NULL,  # can't know age, failed getting creation date
-                                             status,
-                                             FAIL])
-                    raise Exception  # the except block message is accurate in this case
-
-                if vm_creation_time + delta < now:
-                    vm_delta = now - vm_creation_time
-                    with lock:
-                        vms_to_delete[provider_key].add((vm_name, vm_delta))
-            except Exception as e:
-                logger.error(e)
-                logger.error('Failed to get creation/boot time for {} on {}'.format(
-                    vm_name, provider_key))
-                continue
-
-        with lock:
-            print('{} finished processing for matches'.format(provider_key))
+        provider_mgmt = get_mgmt(provider_key)
+        vm_list = provider_mgmt.list_vm()
+
+        # don't really need to thread this, fast enough
+        text_matched_vms = [name for name in vm_list if match(matchers, name)]
+        non_text_matching = set(vm_list) - set(text_matched_vms)
+        print('{}: NOT matching text filters: {}'.format(provider_key, non_text_matching))
+        print('{}: Matching text filters: {}'.format(provider_key, text_matched_vms))
+
+        # MP for vm scanning, could be many VMs per provider
+        # match_queue is passed through and processed by this method's caller
+        non_time_match_queue = Queue()
+        vm_proc_list = []
+        for vm_name in text_matched_vms:
+            vm_proc_list.append(
+                Process(target=scan_vm,
+                        args=(provider_key, vm_name, delta, match_queue,
+                              non_time_match_queue, scan_failure_queue)))
+
+        for proc in vm_proc_list:
+            proc.start()
+        for proc in vm_proc_list:
+            proc.join()
+
+        # each item on scan_queue is a VmData tuple
+        # process non_match_queue and log its contents now
+        non_time_matching = []
+        while not non_time_match_queue.empty():
+            non_time_matching.extend(non_time_match_queue.get())
```

Comment:
could you please explain why you put list with one element everywhere and use extend when retrieve it from queue ?